plugin/pkg/scheduler/api/types.go

```diff
@@ -43,6 +43,7 @@ type PriorityPolicy struct {
 	// For the Kubernetes provided priority functions, the name is the identifier of the pre-defined priority function
 	Name string `json:"name"`
 	// The numeric multiplier for the minion scores that the priority function generates
+	// The weight should be non-zero and can be a positive or a negative integer
```

Comment:
The range is for the score that the priority functions give to each minion and that is correctly stated as 0-10. The weight is allowed to be specified as a negative or positive non-zero integer. The resultant score (after multiplying with the weight) could be any positive/negative integer. 

Actually, the priority functions themselves have no clue about the weight since that is managed outside the priority function (within the PriorityConfig or the PriorityConfigFactory). It is the generic scheduler that multiplies the minion scores with the weight for the given priority function.